### PR TITLE
Add error message to amount box in donation form

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
https://phabricator.wikimedia.org/T182415

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
+1 to Gabriels changes

**Comment by wiese** [comment]
You're on fire, man - will look at and test-drive all that awesomeness first thing tomorrow.

**Comment by JeroenDeDauw** [comment]
The overflow issue does not seem to be fixed yet:

image

File: skins/cat17/templates/Donation_Form.html.twig [hunk 1]
```diff
@@ -53,6 +53,8 @@
 								<div class="wrap-amount-typed field-grp" data-currency="€">
 									<input type="text" id="amount-typed">
 								</div>
+								<span class="error-text">{$ 'form_amount_error' | trans $}</span>
```
Comment (wiese) [inline skins/cat17/templates/Donation_Form.html.twig]: Key still seems to be missing from the content repo.

File: skins/cat17/src/scripts/donationForm.js [hunk 2]
```diff
@@ -138,68 +138,51 @@ $( function () {
 			stateKey: 'donationFormContent.addressType'
 		},
       {
-        viewHandler: WMDE.View.createFieldValueValidityIndicator( $( '#first-name' ) ),
+        viewHandler: WMDE.View.createFieldValueValidityIndicator( $( '.field-firstname' ) ),
```
Comment (wiese) [inline skins/cat17/src/scripts/donationForm.js]: I was very excited to see the ids replaced by class selectors - making half the view handlers redundant in the process. OTOH is adds a certain amount of redundancy (separate attributes only used by one view handler/component each) for now. Would you agree that the general direction, step by step, is to reduce this back to one unique way to identify each input, maybe even reuse elements selected once in the main dispatchers (à la page contract in e.g. `donationForm.js`)

File: skins/cat17/templates/Donation_Form.html.twig (hunk 1 above)
Comment (JeroenDeDauw) [reply skins/cat17/templates/Donation_Form.html.twig]: Added in https://github.com/wmde/fundraising-frontend-content/pull/84

File: skins/cat17/src/scripts/donationForm.js (hunk 2 above)
Comment (gbirke) [reply skins/cat17/src/scripts/donationForm.js]: Yes, that's a direction I'd like to take. Thanks for pointing it out.

**Pull request merged**